### PR TITLE
Feature fe demo user

### PR DESCRIPTION
**Pull request by amber-morning**
### What this PR does (required):
- a demo Login and Sign Up button provided which calls the backend `/auth/demo/login` and `/auth/demo/signup` to load the user and bypass providing data to the form to achieve same functionality

### Any information needed to test this feature (required):
- To see exact login functionality, this feature needs backend to provide  `/auth/demo/login` and `/auth/demo/signup`  endpoints where a logged user object is returned upon post request with application/json as content-type

### CODE REVIEW

File: client/src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -6,6 +6,53 @@ import { useSnackBar } from '../../context/useSnackbarContext';
...
+          updateSnackBarMessage(data.success);
+          updateLoginContext(data.success);
+        } else {
+          console.log({ data });
+          updateSnackBarMessage('unexpected error occurred!');
+        }
+      })
+      .catch(() => ({
+        error: { message: 'Unable to connect to server. Please try again' },
+      }));
```
Comment (moffatethan) [inline client/src/pages/Login/Login.tsx:39]: Could we put this in `helpers/APICalls` like the other API calls in the app 😃

File: client/src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -6,6 +6,53 @@ import { useSnackBar } from '../../context/useSnackbarContext';
...
+    return await fetch(`/auth/demo/login`, fetchOptions)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else if (data.success) {
+          updateSnackBarMessage(data.success);
+          updateLoginContext(data.success);
+        } else {
+          console.log({ data });
```
Comment (moffatethan) [inline client/src/pages/Login/Login.tsx:33]: Remove this console log too please 👍🏻

**Review by moffatethan — CHANGES_REQUESTED** [review]
Super close - good work just want you to remove the log and move the API call to a helper.

File: client/src/pages/Login/Login.tsx (hunk 1 above)
Comment (jtingling) [reply client/src/pages/Login/Login.tsx]: The feature has been reorganized and made more modular.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**